### ISSUE TITLE
Feature request: Add native exfat  character support.

### ISSUE THREAD

**Issue by eyalroz** [linuxmint/nemo#2708]
* Nemo 4.8.6 (built from source, using xapp 2.0.7 and cinnamon-desktop 4.8.1 built from source)
* Windowed (but it doesn't matter)
* Distribution: Devuan Beowulf
* Graphics hardware: Irrelevant
* 64 bit

When I copy a file from a filesystem which supports some set of characters, to a filesystem which supports a different set, and the filename has characters in the difference of the sets - I get an error dialog, which tells me "No such file or directory" about the target filename.

That's the wrong error; and you have to know what's wrong in order to decipher it.

To reproduce:

1. Have an ext4 filesystem mounted
2. Have an exFAT filesystem on a USB-connected HDD, mounted via fuse
3. Create a file named "hello:world" somewhere on the ext4 filesystem
4. Using nemo, try copying this file to a writable location on the mounted filesystem. 

The kernel version may also be playing a part here, so if you're not on the exact same distro as me, try at least to be using Linux 5.9.0.

This should get you the invalid/misleading dialog mentioned above.

**Comment by leigh123linux** [comment]
nemo-3.8.x is EOL, feel free to reopen if you can reproduce on a supported version.

**Comment by eyalroz** [comment]
@leigh123linux : I can't reopen myself, only project members can. please reopen.

**Comment by leigh123linux** [comment]
You will need to provide more info, please provide full steps to reproduce.

**Comment by eyalroz** [comment]
@leigh123linux : See edit, but it's not as though I'm talking about some esoteric scenario.

**Comment by leigh123linux** [comment]
I can't reproduce to issue, hello:world is automatically renamed to hello_world when copying from ext4 to fat.

https://github.com/linuxmint/nemo/blob/master/libnemo-private/nemo-file-operations.c#L3392

**Comment by eyalroz** [comment]
@leigh123linux : Interesting! I've constricted the reproduction instructions.

Also, the automatic-renaming is another reason why this bug and #2709 are different :-)

**Comment by leigh123linux** [comment]
I can reproduce the issue with native exfat, it looks like glib will need exfat fix 

https://gitlab.gnome.org/GNOME/nautilus/-/merge_requests/565#note_847645

https://github.com/GNOME/glib/blob/46097e9b8e4ad253e0e3320e929cc0fa3df84d5d/gio/glocalfile.c#L563

**Comment by eyalroz** [comment]
@leigh123linux : 

In the mean time, I believe Nemo should check for characters unsupported in the target filesystem.

In the longer run, there will remain the question of #2709 (which would require such a check to remain, so as to be able to implement a different policy than glib's auto-replacement with underscore).

**Comment by leigh123linux** [comment]
Depends on glib adding support, see bug report https://gitlab.gnome.org/GNOME/glib/-/issues/2369

**Comment by eyalroz** [comment]
@leigh123linux : Then let me rephrase my mean-time suggestion: 

When:

1. The target filesystem is not known, and 
2. A file copy fails
3. The filename contains characters outside of a "least common denominator" set (e.g. when it contains colons, double-quotation marks and other character which are unsupported by at least one commonly-encountered filesystems)

Then either:

1. Offer to convert the questionable characters to underscores (in the failure dialog or as an apriori option in Edit|Preferences), or
2. Mention the possibility of this being the cause of failure in the error notification dialog

**Comment by eyalroz** [comment]
@leigh123linux : I can understand how you feel personally, but:

1. Even if we dislike MS operating systems and/or file systems, many of nemo's users don't; or they have to interact with people who don't.
2. HDDs and SSDs often come pre-formatted with exFAT.
3. (Smaller) USB flash drives often come pre-formatted with exFAT. Some may even be formatted in plain FAT32.

Also, acting only based on glib support will mean only future OS releases will be able to enjoy a fix for this situation, while remedial action as things stand will allow existing OS distributions to update their nemo version and address this issue.

I realize it is not a fun request to have write code whose main use is addressing a deficiency in a current library, but it would still help users in a common scenario.

**Comment by eyalroz** [comment]
The new PR arranges for auto-renaming to `_` of unsupported characters.
